### DISCUSSION TITLE
Kamal Infrastructure

### DISCUSSION THREAD

**Discussion by AlyonaChm** [basecamp/kamal#938] [Q&A] [answered]
Hi all,
Do you know where I can find a Kamal infrastructure schema or components for a better understanding of how it works? I’m looking for something similar to this Kubernetes overview https://kubernetes.io/docs/concepts/overview/components/

**Comment by dhh** [comment, marked as answer]
There's no such overview, but would be great to have one. It would be quite simple, though. It just uses docker directly and then sticks Traefik in front of it (for v1) or Kamal Proxy (for v2) to allow zero-downtime deploys. Would make for a pretty boring diagram 😄
